pkg/ucp/credentials/azure.go

```diff
@@ -69,10 +69,17 @@ func (p *AzureCredentialProvider) Fetch(ctx context.Context, planeName, name str
 		case *ucpapi.InternalCredentialStorageProperties:
 			storage = c
 		default:
-			return nil, errors.New("invalid AzureServicePrincipalProperties")
+			return nil, errors.New("invalid InternalCredentialStorageProperties")
+		}
+	case *ucpapi.AzureWorkloadIdentityProperties:
+		switch c := p.Storage.(type) {
+		case *ucpapi.InternalCredentialStorageProperties:
+			storage = c
+		default:
+			return nil, errors.New("invalid InternalCredentialStorageProperties")
 		}
 	default:
-		return nil, errors.New("invalid InternalCredentialStorageProperties")
+		return nil, errors.New("invalid AzureCredentialProperties")
```

Comment:
Is this error customer facing?

Reply:
see above

Reply:
Don't we need to add better error messages here? I know this was always the case but should we udate these messages?